modules/Emails/include/ListView/ListViewDataEmails.php

```diff
@@ -269,209 +191,817 @@ public function getListViewData(
-                        case 'uid':
-                            $emailRecord[strtoupper($field)] = $emailHeader['uid'];
-                            $_REQUEST['email_uids'][] = $emailHeader['uid'];
-                            break;
-                        case 'msgno':
-                            $emailRecord[strtoupper($field)] = $emailHeader['msgno'];
-                            break;
-                        case 'has_attachment':
-                            $emailRecord[strtoupper($field)] = $emailHeader['has_attachment'];
-                            break;
-                        case 'status':
-                            if($emailHeader['answered'] != 0) {
-                                $emailRecord[strtoupper($field)] = 'replied';
-                            } else if($emailHeader['draft'] != 0) {
-                                $emailRecord[strtoupper($field)] = 'draft';
-                            } else if($emailHeader['seen'] != 0) {
-                                $emailRecord[strtoupper($field)] = 'read';
-                            } else {
-                                $emailRecord[strtoupper($field)] = 'unread';
+                            if(isset($post_retrieve)) {
+                                $parent_fields = $seed->retrieve_parent_fields($post_retrieve);
+                                foreach($parent_fields as $child_id => $parent_data) {
+                                    //add it to every row with the given id
+                                    foreach($idIndex[$child_id] as $index){
+                                        $rows[$index]['parent_name']= $parent_data['parent_name'];
+                                    }
+                                }
                             }
+                        }
+
+                        $pageData = array();
 
-                            if($emailHeader['deleted'] != 0) {
-                                // TODO: TASK: UNDEFINED - Handle deleted
+                        reset($rows);
+                        while($row = current($rows)){
 
+                            $temp = clone $seed;
+                            $dataIndex = count($data);
+
+                            $temp->setupCustomFields($temp->module_dir);
+                            $temp->loadFromRow($row);
+                            if (empty($this->seed->assigned_user_id) && !empty($temp->assigned_user_id)) {
+                                $this->seed->assigned_user_id = $temp->assigned_user_id;
+                            }
+                            if($idIndex[$row[$id_field]][0] == $dataIndex){
+                                $pageData['tag'][$dataIndex] = $temp->listviewACLHelper();
+                            }else{
+                                $pageData['tag'][$dataIndex] = $pageData['tag'][$idIndex[$row[$id_field]][0]];
+                            }
+                            $data[$dataIndex] = $temp->get_list_view_data($filter_fields);
```

Comment:
SugarBean needs refactored. The in the Contact.php the Contact Bean has redefined this method call. We will need to look at this in a future release. For now I have removed the argument since I don't need in it for emails